labs-ios-starter/Map/LocationSearchTableViewController.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  LocationSearchTableViewController.swift
+//  labs-ios-starter
+//
+//  Created by Claudia Maciel on 1/20/21.
+//  Copyright © 2021 Spencer Curtis. All rights reserved.
+//
+
+import UIKit
+import MapKit
+
+
+class LocationSearchTableViewController: UITableViewController {
+
+    var matchingItems:[MKMapItem] = []
+    var mapView: MKMapView? = nil
+    
+    var handleMapSearchDelegate: HandleMapSearch? = nil
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+    }
+    
+    //MARK: - Show Address on TableView
+    func parseAddress(selectedItem:MKPlacemark) -> String {
+        let firstSpace = (selectedItem.subThoroughfare != nil && selectedItem.thoroughfare != nil) ? " " : ""
+        let comma = (selectedItem.subThoroughfare != nil || selectedItem.thoroughfare != nil) && (selectedItem.subAdministrativeArea != nil || selectedItem.administrativeArea != nil) ? ", " : ""
+        let secondSpace = (selectedItem.subAdministrativeArea != nil && selectedItem.administrativeArea != nil) ? " " : ""
+        let addressLine = String(
+            format:"%@%@%@%@%@%@%@",
+            // street number
+            selectedItem.subThoroughfare ?? "",
+            firstSpace,
+            // street name
+            selectedItem.thoroughfare ?? "",
+            comma,
+            // city
+            selectedItem.locality ?? "",
+            secondSpace,
+            // state
+            selectedItem.administrativeArea ?? ""
+        )
+        return addressLine
+    }
```

Comment:
This is pretty confusing code, uses far too much nil coalescing, and in the event that you are localizing the app for other languages, wouldn't translate very well. I would consider putting all of the things that you do have inside of an array, and then using the `joined()` method with a comma + space separator